azurerm/data_source_arm_scheduled_time_test.go

```diff
@@ -0,0 +1,444 @@
+					resource.TestCheckResourceAttr(dataSourceName, "frequency", "Week"),
+					resource.TestCheckResourceAttr(dataSourceName, "next_run_time", formattedExpectedTime),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMScheduledTime_Weekly_Next_Week(t *testing.T) {
+	dataSourceName := "data.azurerm_scheduled_time.test"
+
+	now := time.Now().UTC()
+	scheduletime := now.Add(time.Duration(168) * time.Hour) //1 week
+	dayofweek := int(scheduletime.Weekday())
+
+	config := testAccDataSourceAzureRMScheduledTime_Weekly(dayofweek, scheduletime)
+
+	expectedTime := time.Date(scheduletime.Year(), scheduletime.Month(), scheduletime.Day(), scheduletime.Hour(), 0, 0, 0, time.UTC)
+	formattedExpectedTime := expectedTime.Format(time.RFC3339)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "day_of_week", strconv.Itoa(dayofweek)),
+					resource.TestCheckResourceAttr(dataSourceName, "hour", strconv.Itoa(now.Hour())),
+					resource.TestCheckResourceAttr(dataSourceName, "minute", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "second", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "frequency", "Week"),
+					resource.TestCheckResourceAttr(dataSourceName, "next_run_time", formattedExpectedTime),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMScheduledTime_Monthly_Tomorrow(t *testing.T) {
+	dataSourceName := "data.azurerm_scheduled_time.test"
+
+	now := time.Now().UTC()
+	scheduletime := now.Add(time.Duration(24) * time.Hour) //1 day
+	dayofmonth := scheduletime.Day()
+
+	config := testAccDataSourceAzureRMScheduledTime_Monthly(dayofmonth, scheduletime)
+
+	expectedTime := time.Date(scheduletime.Year(), scheduletime.Month(), scheduletime.Day(), scheduletime.Hour(), 0, 0, 0, time.UTC)
+	formattedExpectedTime := expectedTime.Format(time.RFC3339)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "day_of_month", strconv.Itoa(dayofmonth)),
+					resource.TestCheckResourceAttr(dataSourceName, "hour", strconv.Itoa(now.Hour())),
+					resource.TestCheckResourceAttr(dataSourceName, "minute", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "second", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "frequency", "Month"),
+					resource.TestCheckResourceAttr(dataSourceName, "next_run_time", formattedExpectedTime),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMScheduledTime_Monthly_Next_Month(t *testing.T) {
+	dataSourceName := "data.azurerm_scheduled_time.test"
+
+	now := time.Now().UTC()
+	scheduletime := now.Add(25 * time.Duration(24) * time.Hour) //25 days
```

Comment:
(as below) I think it'd be better to construct the date and add a month here - that would remove the need for the `if` statement?